packages/new-polymath-issuer/src/pages/DividendsWizard/Container.tsx

```diff
@@ -203,48 +208,118 @@ export class ContainerBase extends Component<Props, State> {
     const parsedCheckpointIndex = parseInt(checkpointIndex, 10);
     return (
       <Page title="Create New Dividend Distribution">
         <DataFetcher
           watchProps={this.state}
-          fetchers={[
-            createTaxWithholdingListBySymbolAndCheckpointFetcher({
-              securityTokenSymbol,
-              checkpointIndex: parsedCheckpointIndex,
-              dividendType: DividendModuleTypes.Erc20,
-            }),
-            createCheckpointBySymbolAndIdFetcher({
-              securityTokenSymbol,
-              checkpointIndex: parsedCheckpointIndex,
-            }),
-          ]}
-          render={(
-            {
-              taxWithholdings,
-              checkpoints: [checkpoint],
-            }: {
-              taxWithholdings: types.TaxWithholdingEntity[];
-              checkpoints: types.CheckpointEntity[];
-            },
-            loading: boolean
-          ) => {
+          fetchers={[fetcher]}
+          render={(data: { checkpoints: types.CheckpointEntity[] }) => {
+            const { checkpoints } = data;
+            const fetchers = checkpoints.map(({ index }) =>
+              createDividendsByCheckpointFetcher(
+                {
+                  securityTokenSymbol,
+                  checkpointIndex: index,
+                },
+                { propKey: `${index}` }
+              )
+            );
             return (
-              <Presenter
-                createDividendDistribution={this.createDividendDistribution}
-                updateTaxWithholdingList={this.updateTaxWithholdingList}
-                stepIndex={step}
-                securityTokenSymbol={securityTokenSymbol}
-                checkpoint={checkpoint}
-                onNextStep={this.nextStep}
-                onPreviousStep={this.previousStep}
-                taxWithholdings={taxWithholdings}
-                downloadTaxWithholdingList={this.downloadTaxWithholdingList}
-                downloadSampleExclusionList={this.downloadSampleExclusionList}
-                fetchBalance={this.fetchBalance}
-                fetchIsValidToken={this.fetchIsValidToken}
-                isLoadingData={loading}
+              <DataFetcher
+                watchProps={this.state}
+                fetchers={fetchers}
+                render={(dividendsData: {
+                  [key: string]: types.DividendEntity[];
+                }) => {
+                  const allDividendsCompleted = every(
+                    flatten(
+                      map(dividendsData, dividends =>
+                        map(dividends, dividend => {
+                          const { expiry, investors } = dividend;
+
+                          const remainingPayments = investors.filter(
+                            (investor: any) =>
+                              !investor.paymentReceived && !investor.excluded
+                          ).length;
+
+                          const remainingTransactions = Math.ceil(
```

Comment:
Calculating this value is unnecessary (so importing `DIVIDEND_PAYMENT_INVESTOR_BATCH_SIZE` is unnecessary as well). As long as there is 1 remaining payment, we should redirect)